### PR TITLE
Rename EditorTilesetFilter to MapEditorData

### LINKED ISSUE #13857: Rename EditorTilesetFilter to something more generic
From IRC:
```
[17:17] <reaperrr> pchote: Would removing 'Tileset' from the trait name be enough ('EditorFilter'), or would 'EditorVisibility' be better?
[17:18] <+pchote> I would prefer something like MapEditorData
```

### PR DESCRIPTION
**Pull request by Ectras**
Closes #13857.
Renamed the EditorTilesetFilter class to MapEditorData (and EditorTilesetFilterInfo to MapEditorDataInfo).
Created an UpdateRule (RenameEditorTilesetFilter.cs). Updated all four mods with it.

### CODE REVIEW

**Comment by Ectras** [comment]
Moved the UpdateRule to the Rules folder, filtered out the .ormap files

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj [hunk 1]
```diff
@@ -919,22 +920,14 @@
...
       mcs generates .dll.mdb symbol files.
     -->
-    <Copy
-      SourceFiles="$(TargetDir)$(TargetName).pdb"
-      DestinationFolder="$(SolutionDir)mods/common/"
-      Condition="Exists('$(TargetDir)$(TargetName).pdb')"/>
-    <Copy
-      SourceFiles="$(TargetPath).mdb"
-      DestinationFolder="$(SolutionDir)mods/common/"
-      Condition="Exists('$(TargetPath).mdb')"/>
```
Comment (Mailaender) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: Please don't commit these changes.

**Comment by Ectras** [comment]
I'm really sorry for that. I committed the original file now (with just the necessary changes). Hope that's ok.

File: OpenRA.Mods.Common/UpdateRules/Rules/RenameEditorTilesetFilter.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		{
+			get
+			{
+				return "The name EditorTilesetFilter was misleading and was renamed to MapEditorData.";
+			}
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			foreach (var spawner in actorNode.ChildrenMatching("EditorTilesetFilterInfo"))
```
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/Rules/RenameEditorTilesetFilter.cs]: The `Info` classes are an internal implementation detail - they aren't exposed to the yaml, so you don't need to worry about them here :)

File: OpenRA.Mods.Common/UpdateRules/Rules/RenameEditorTilesetFilter.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+				spawner.RenameKey("MapEditorDataInfo");
+
+			foreach (var spawner in actorNode.ChildrenMatching("EditorTilesetFilter"))
```
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/Rules/RenameEditorTilesetFilter.cs]: Please rename this variable to `editor` or `data` or something similar (copy/paste leftover from `WormSpawner`).

**Review by pchote — COMMENTED** [review]
Code changes look reasonable aside from two minor nits.  I'm not able to test this ingame.

**Comment by pchote** [comment]
We'll also want you to squash the commits together once all the fixups are sorted

**Comment by Ectras** [comment]
I feel a bit like a fool right now. Well, once again, really sorry. Fixed the two mistakes and squashed everything into one commit. Hope I didn't miss anything this time.

**Review by teinarss — APPROVED** [review]
I did some testing: 
* RenameEditorTilesetFilter works as intended
* No problems found in ActorSelectorLogic

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1254a393e5e50a54042866f406beccef39b32575...17205547d729143b585f552c4c572afdb66590c1)

**Pull request merged**